### ISSUE TITLE
ReadLocalValue() is never returning UnsetValue

### ISSUE THREAD

**Issue by ajpinedam** [unoplatform/uno#18531]
### Current behavior

There are a few places in the Uno code where we use `ReadLocalValue` to validate whether a `DependencyProperty` has been set, returning `UnsetValue`. I have seen situations where we are returning null for DPs that still need to be set, and this behavior change is causing some issues.

### Workaround

N/A

### Works on UWP/WinUI

None

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

5.6.0-dev.149 

### Affected platforms

WebAssembly, Android, iOS, Mac Catalyst, Skia (WPF), Skia (Linux X11), Skia (macOS), Skia (Linux Framebuffer), Skia (GTK)

### IDE

Visual Studio 2022, Visual Studio Code

**Comment by ajpinedam** [comment]
cc: @Youssef1313  I saw this issue while debugging a fix on PR https://github.com/unoplatform/uno/pull/18492

**Comment by Youssef1313** [comment]
That's quite strange. The implementation is currently:

https://github.com/unoplatform/uno/blob/b6333bd85ef09975ae45e538230dae337a96e43f/src/Uno.UI/UI/Xaml/DependencyObjectStore.cs#L281-L295

**Comment by ajpinedam** [comment]
> if (details?.GetBaseValueSource() == DependencyPropertyValuePrecedences.Local)

While debugging the code entered here

```chsarp
if (details?.GetBaseValueSource() == DependencyPropertyValuePrecedences.Local) 
{ 
    return details.GetBaseValue(); 
} 
```

and 

```csharpreturn details.GetBaseValue(); 
```

is returning null instead of `UnserValue`

cc: @Youssef1313

**Comment by Youssef1313** [comment]
That should mean that a local value of null was set.
Are you able to track the SetValue calls?

**Comment by ajpinedam** [comment]
This is for the `SelectedItem` on the `Selector`. I set a breakpoint inside the `Set`, which is never hit before hitting this method call. 

We can get some time tomorrow and do a quick sync whenever you're available to look into it

cc: @Youssef1313

**Comment by MartinZikmund** [comment]
It could be also set not via `set` block, but via `SetValue` for example

**Comment by ajpinedam** [comment]
> It could be also set not via `set` block, but via `SetValue` for example. Youssef may be busy, but I can connect with you on this @ajpinedam

Set a conditional breakpoint in the `SetValue`, and it only gets hit after the method is called. Let's schedule a quick pairing session tomorrow.

**Comment by MartinZikmund** [comment]
@ajpinedam To reproduce I set the breakpoint in `SetValueInternal` method, and made it conditional (e.g. `property == TabView.SelectedItem`)

**Comment by Youssef1313** [comment]
@MartinZikmund @ajpinedam Was it that something is actually setting the DP to local value?

**Comment by MartinZikmund** [comment]
@Youssef1313 yes, the test sets data context which in turn sets the `SelectorItem` bound value to `null` (there is a two-way binding for it). Not sure yet why the behavior differs from WinUI though
